bpf/process/bpf_execve_event.c

```diff
@@ -124,8 +124,18 @@ event_filename_builder(void *ctx, struct msg_process *curr, __u32 curr_pid, __u3
 			flags |= EVENT_DATA_FILENAME;
 		}
 	}
+
 	curr->flags = flags;
+	/* Get curr_pid from argument to avoid BPF verifier errors and
+	 * use it for both pid and tid.
+	 * During an execve all threads other than the calling thread
+	 * are destroyed, but since we hook late during the execve then
+	 * the calling thread at the hook time is already the new thread
+	 * group leader.
+	 */
 	curr->pid = curr_pid;
+	curr->tid = curr_pid;
```

Comment:
do we really need tid field in msg_process if it will be always pid?
same for clone and exit events.. IIUC we record tid for generic kprobes/uprobes/tracepoints
and then set it to instance of user space process  event.. but tid is never stored, right?

Reply:
> do we really need tid field in msg_process if it will be always pid? same for clone and exit events..
 
Ok so to break it:

- Clone we don't export events, only from ringbuffer to user space cache, so we can do `tid=pid` at the userspace, I thought of it and for the sake of asserting our current design, especially since: a. clone can create threads, and b. we are fixing stuff now since it was overlooked, I prefer to have it set in BPF as it is clear and covered with ring buffer tests.

- For exec and exit event, no strong opinion, since right now we hook late at the exec, so a. we can remove the tid from msg_process and set it at user space, or b. send it part of the BPF to be clear and consistent?
In future in case we do binary digest, we may hook early to catch interpreter (bash, etc), in that case if tid != pid then we may rethink this...  

> IIUC we record tid for generic kprobes/uprobes/tracepoints
and then set it to instance of user space process event.. but tid is never stored, right?

Yes tid is recorded on the fly at bpf for kprobes/uprobes/tracepoints and then we set it in user space, since the process in userspace in the cache that came from execve or clone should have `pid==tid`,  so `tid` is stored in the user space cache but it should always equal `pid`.

For why it is stored, no choice since it is part the [process api](https://github.com/cilium/tetragon/blob/main/pkg/process/process.go#L42) inside of `ProcessInternal`